### ISSUE TITLE
Fix Request: false positive in pmd:AvoidExposingMutableRecordState

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#628]
Nested records are not supported yet.
Example:
```java
public record ApprovalNotificationProperties(
        Email email,
        URI linkUrl
) {
    record Email(
            boolean enabled,
            Set<String> specificIds
    ) {
        Email {
            specificIds = Set.copyOf(specificIds.stream()
                    .map(String::strip)
                    .filter(Predicate.not(String::isEmpty))
                    .collect(Collectors.toSet()));
        }
    }
}
```

**Comment by jborgers** [comment]
Was already fixed with #554
